### PR TITLE
Adds pre-commit configuration (& markdownlint-cli2)

### LINKED ISSUE #5: Add pre-commit and linting
It would be useful/beneficial to apply the same suite of [pre-commit](https://pre-commit.com) hooks to this repository
as we do to TopoStats to ensure consistent coding styles and standards. Similarly integrating
[pre-commit.ci](https://pre-commit.ci) to the Continuous Integration would also make sense.

### PR DESCRIPTION
**Pull request by ns-rse**
Closes #5

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -18,11 +18,106 @@ classifiers = [
     "Programming Language :: Python :: 3.9",
     "Programming Language :: Python :: 3.10",
 ]
+keywrods = [
```
Comment (SylviaWhittle) [inline pyproject.toml:21]:
```suggestion
keywords = [
```

**Review by SylviaWhittle — CHANGES_REQUESTED** [review]
Thank you so much for looking over this repo and starting to bring it up to par with TopoStats. 😄 

Just one typo I noticed, sorry!

File: pyproject.toml (hunk 1 above)
Comment (ns-rse) [reply pyproject.toml:21]: Good catch thanks, my fingers were clearly not warmed up that early in the morning.

**Review by SylviaWhittle — APPROVED** [review]
Thank you 😄

**Pull request merged**